index/matchtree.go

```diff
@@ -1201,19 +1225,28 @@ func (d *indexData) newMatchTree(q query.Q, opt matchTreeOpt) (matchTree, error)
 		}, nil
 
 	case *query.RepoIDs:
+		checksum := queryRepoIdsChecksum(d.repoMetaData)
+		cacheKeyField := "RepoIDs"
+		if cached, ok := d.docMatchTreeCache.get(cacheKeyField, checksum); ok {
+			return cached, nil
+		}
```

Comment:
if repoids didn't show up in your profiling/testing, can we avoid caching in it? IE avoid caching unless its shown to be effective.

If it did show up, do you mind experimenting with what I proposed around updating the predicate we use to be smarter (ie avoid the intermediate reposWant slice)

Reply:
> if repoids didn't show up in your profiling/testing, can we avoid caching in it? IE avoid caching unless its shown to be effective.

Sure, that makes sense. I went ahead and removed `repoids` from this PR. Happy to circle back with another iteration of this that includes `repoids` if we can report a performance improvement. 🤝